### PR TITLE
Add test where units are expected to be linked after parsing a model

### LINKED ISSUE #887: Units not linking or not reported as linked when parsing a model
Contrary to my understanding when I created a test for the `Analyser` I thought the `Parser` would link all the Units.  My test showed that either this wasn't the case or the `hasUnlinkedUnits` method doesn't give the correct answer.

### PR DESCRIPTION
**Pull request by hsorby**
Closes #887.

### CODE REVIEW

File: tests/parser/file_parser.cpp [hunk 1]
```diff
@@ -162,3 +162,14 @@ TEST(Parser, parseModelWithImportedEquivVariables)
 
     EXPECT_EQ(modelContents, serialisedModel);
 }
+
+TEST(Parser, unitsAreLinkedOnParse)
+{
+    libcellml::ParserPtr p = libcellml::Parser::create();
+    auto model = p->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    printModel(model, false);
```
Comment (agarny) [inline tests/parser/file_parser.cpp]: Debug code, so remove it.

**Review by agarny — CHANGES_REQUESTED** [review]

File: tests/parser/file_parser.cpp (hunk 1 above)
Comment (hsorby) [reply tests/parser/file_parser.cpp]: 8e16b79

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Comment by nickerso** [comment]
@hsorby - the error in the coverage looks like the buildbot might have run out of space?

**Comment by hsorby** [comment]
I don't think so, I think that is a disk failed error.  I think this is due to an error with the virtual disk more than anything.  I'll try a rebuild.

**Pull request merged**